cli/command_server_start.go

```diff
@@ -217,25 +218,28 @@ func (c *commandServerStart) run(ctx context.Context) error {
 			return errors.Wrap(httpServer.Close(), "close")
 		}
 
+		rep := srv.GetRepository()
+		if rep != nil {
+			//nolint:errcheck
+			rep.Close(ctx)
+		}
+
 		log(ctx2).Debugf("graceful shutdown succeeded")
 
 		return nil
 	}
 
-	c.svc.onCtrlC(func() {
-		log(ctx).Infof("Shutting down...")
+	c.svc.onCtrlC(func() { shutdownServer(ctx, httpServer, srv) })
 
-		if serr := httpServer.Shutdown(ctx); serr != nil {
-			log(ctx).Debugf("unable to shut down: %v", serr)
-		}
-	})
+	c.svc.onSigTerm(func() { shutdownServer(ctx, httpServer, srv) })
 
-	c.svc.onRepositoryFatalError(func(_ error) {
-		if serr := httpServer.Shutdown(ctx); serr != nil {
-			log(ctx).Debugf("unable to shut down: %v", serr)
-		}
+	c.svc.onSigDump(func() {
+		debug.StopProfileBuffers(ctx)
```

Comment:
Maybe log right before dumping to give a feedback that we got the signal ?

Reply:
@Shrekster feedback is give within `StopProfileBuffers`